src/components/PersonList/PersonList.tsx

```diff
@@ -0,0 +1,94 @@
+import { FC } from 'react';
+import classNames from 'classnames';
+import { useParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink/PersonLink';
+
+type Props = {
+  persons: null | Person[],
+};
+
+export const PersonList: FC<Props> = ({ persons }) => {
+  const { slug } = useParams();
+
+  if (!persons) {
+    return (
+      <p data-cy="noPeopleMessage">
+        {persons === null
+          ? ''
+          : 'There are no people on the server'}
+      </p>
+    );
+  }
+
+  return (
+    <table
+      data-cy="peopleTable"
+      className="table is-striped is-hoverable is-narrow is-fullwidth"
+    >
+      <thead>
+        <tr>
+          <th>Name</th>
+          <th>Sex</th>
+          <th>Born</th>
+          <th>Died</th>
+          <th>Mother</th>
+          <th>Father</th>
+        </tr>
+      </thead>
+
+      <tbody>
+        {persons.map(person => {
+          const {
+            name,
+            sex,
+            born,
+            died,
+            motherName,
+            fatherName,
+          } = person;
+
+          const findPerson = (personName: string) => {
+            const foundPerson = persons.find(pers => pers.name === personName);
+
+            if (foundPerson) {
+              return (
+                <PersonLink
+                  slug={foundPerson.slug}
+                  sex={foundPerson.sex}
+                  title={foundPerson.name}
+                />
+              );
+            }
+
+            return personName;
+          };
```

Comment:
Now everything works correctly, but it is better to carry out large logical calculations outside the rendering and use the ready-made result. Right away, such code is hard to read and maintain.